app/src/main/java/android/coding/interview/makeitawesome/fragment/PicturesFragment.java

```diff
@@ -19,19 +30,78 @@
 public class PicturesFragment extends Fragment {
 
     public static Fragment newInstance() {
-        return new PicturesFragment();
+        PicturesFragment pictures = new PicturesFragment();
+        pictures.setRetainInstance(true);
+        return pictures;
     }
 
+    RecyclerView rv;
+    ProgressDialog mProgressDialog;
+    List<Album> mData;
+    boolean loading = false;
+    private TinyBus mBus;
+
     @Nullable
     @Override
     public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
-        RecyclerView rv = (RecyclerView) inflater.inflate(R.layout.pictures_list_fragment, container, false);
-        setupRecyclerView(rv);
+        rv = (RecyclerView) inflater.inflate(R.layout.pictures_list_fragment, container, false);
+
+        if (mProgressDialog != null && mProgressDialog.isShowing())
+            mProgressDialog.dismiss();
+
+        mProgressDialog = new ProgressDialog(container.getContext());
+        mProgressDialog.setIndeterminate(true);
+        mProgressDialog.setMessage("Loading...");
+        mProgressDialog.show();
```

Comment:
All this code is messy. He should have at least encapsulated some parts in functions.